src/main/java/randoop/generation/OrienteeringSelection.java

```diff
@@ -0,0 +1,244 @@
+  /**
+   * Bias input selection towards lower cost sequences. We first compute and update the weights of
+   * all the candidates within the candidate list before making our selection.
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  @Override
+  public Sequence selectInputSequence(SimpleList<Sequence> candidates) {
+    double totalWeight = updateWeightMapForCandidates(candidates);
+
+    Sequence selectedSequence = Randomness.randomMemberWeighted(candidates, weightMap, totalWeight);
+    incrementCountInMap(sequenceSelectionCount, selectedSequence);
+    return selectedSequence;
+  }
+
+  /**
+   * Update the weights of the candidates in the given list. If an input sequence has not been
+   * selected or executed before, it will be assigned its default weight, computed by {@code
+   * Sequence.getSize()}. The formula for updating a sequence's weight if selection count and
+   * execution time information are both available is
+   *
+   * <p>1.0 / (k * seq.exec_time * sqrt(seq.meth_size))
+   *
+   * <p>Where k is the number of selections of seq and exec_time is the execution time of seq and
+   * meth_size is the number of method call statements in seq. This formula is a slight
+   * simplification of the one described in the GRT paper which maintains a separate exec_time for
+   * each execution of seq. However, we assume that the execution times of a sequence are the same
+   * as the first execution.
+   *
+   * @param candidates list of candidate sequences
+   * @return the total weight of all the elements in the candidate list
+   */
+  private double updateWeightMapForCandidates(SimpleList<Sequence> candidates) {
+    double totalWeight = 0.0;
+
+    for (int i = 0; i < candidates.size(); i++) {
+      Sequence candidate = candidates.get(i);
+
+      double methodSizeSqrt = getMethodSizeSquareRootForSequence(candidate);
+      Integer selectionCount = sequenceSelectionCount.get(candidate);
+      Long executionTime = sequenceExecutionTime.get(candidate);
+
+      double weight;
+      // Recompute and update this sequence's weight. Note that we check methodSizeSqrt is not zero
+      // to prevent division by zero. This could occur for a sequence that has no method calls.
+      if (selectionCount != null && executionTime != null && methodSizeSqrt != 0) {
+        weight = 1.0 / (selectionCount * executionTime * methodSizeSqrt);
+      } else {
+        weight = candidate.getWeight();
+      }
+
+      totalWeight += weight;
+      weightMap.put(candidate, weight);
+    }
+
+    return totalWeight;
+  }
+
+  /**
+   * Each {@link Sequence} within {@code inputSequences} is a subsequence of {@code eSeq}. At this
+   * point, the given {@link ExecutableSequence} has been executed and contains the execution time
+   * of the sequence as a whole. The ExecutableSequence, eSeq, contains an {@link
+   * randoop.sequence.Execution} which contains a list of {@link ExecutionOutcome}s. The execution
```

Comment:
Is this still necessary?  I thought we discussed that were going to use sequence weights and would never need to do a computation that summed individual statement execution times.  Did that turn out to be a bad idea?  Or have you just not implemented it yet?

Reply:
Summing up the individual execution times is not necessary - I have not yet implemented this part yet. I was briefly looking through what changes would need to be made and it seems quite substantial. I was thinking that a separate pull request would be necessary.